runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -211,6 +211,372 @@ doubleMaxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMaxOp)
+
+   TR::Register* v_s1Partial       = cg->allocateRegister(TR_VRF);
+   TR::Register* v_s2Partial       = cg->allocateRegister(TR_VRF);
+   TR::Register* v_s2Char1Rep      = cg->allocateRegister(TR_VRF);
+   TR::Register* v_tmp             = cg->allocateRegister(TR_VRF);
+
+   // register dependencies
+   TR::RegisterDependencyConditions * regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 17, cg);
+   regDeps->addPostCondition(r_s1Value  , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_s1Len    , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_s2Value  , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_s2Len    , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_fromIndex, TR::RealRegister::AssignAny);
+
+   regDeps->addPostCondition(r_resultReg      , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_maxIndex       , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_s1VecStartIndex, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_s2VecStartIndex, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_loadLen        , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_tmp            , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_tmp2           , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_1stCharIndex   , TR::RealRegister::AssignAny);
+
+   regDeps->addPostCondition(v_s1Partial , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(v_s2Partial , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(v_s2Char1Rep, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(v_tmp       , TR::RealRegister::AssignAny);
+
+   // Labels
+   TR::LabelSymbol* label_find_1st_char        = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_load_len_16          = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_load_len_done        = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_extract_1st_char_pos = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_match_s2_loop        = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_match_s2_loop_prep   = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_s2_partial_match     = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_load_result          = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_done                 = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+
+   // Ops
+   TR::InstOpCode::Mnemonic AHI  = is64Bit ? TR::InstOpCode::AGHI : TR::InstOpCode::AHI;
+   TR::InstOpCode::Mnemonic AHIK = is64Bit ? TR::InstOpCode::AGHIK : TR::InstOpCode::AHIK;
+   TR::InstOpCode::Mnemonic ARK  = is64Bit ? TR::InstOpCode::AGRK : TR::InstOpCode::ARK;
+   TR::InstOpCode::Mnemonic AR   = is64Bit ? TR::InstOpCode::AGR : TR::InstOpCode::AR;
```

Comment:
We should be using the `TR::InstOpCode` APIs for these. For example this one should be `TR::InstOpCode::getAddRegOpCode()`. I suggest inlining them as well to keep symmetry with all other evaluators.

Reply:
Using inlined `TR::InstOpCode` getters everywhere.